Assets/Scripts/Menus/SettingsMenu.cs

```diff
@@ -15,16 +15,37 @@ public class SettingsMenu : MonoBehaviour
     public Slider musicSlider; // The slider that controls music volume.
     public Slider SFXSlider; // The slider that controls sound effect volume.
 
-    private void Start()
+    public Slider mouseSensSlider;
+    public static float mouseSens;
+
+    public GameObject parent;
+    private void Awake()
     {
         musicSlider.value = PlayerPrefs.GetFloat("musicVolume", .5f); // Gets the float value of musicVolume, or uses .5f if it isn't found.
+        musicMixer.SetFloat("musicVolume", musicSlider.value);
+
         SFXSlider.value = PlayerPrefs.GetFloat("SFXVolume", .5f); // Gets the float value of SFXVolume, or uses .5f if it isn't found.
+        SFXMixer.SetFloat("SFXVolume", SFXSlider.value);
+
+        //Set mouse settings
+        mouseSensSlider.value = PlayerPrefs.GetFloat("mouseSens", .5f);
+        mouseSens = PlayerPrefs.GetFloat("mouseSens", .5f);
+
+        //Disable self after initialising settings
+        if (parent != null)
+            transform.SetParent(parent.transform);
+
+        gameObject.SetActive(false);
     }
+ 
 
     public void SetMusicVolume(float musicVol)
-    {
-        musicMixer.SetFloat("musicVolume", Mathf.Log10(musicVol) * 20);
+    {        
         PlayerPrefs.SetFloat("musicVolume", musicVol); // Sets the value of SliderVolume to the music volume value.
+        musicMixer.SetFloat("musicVolume", Mathf.Log10(musicVol) * 20);
+
+        if (musicVol <= 0)
+            musicMixer.SetFloat("musicVolume", 0); //Avoid mathematical errors with Mathf.Log10(0);
```

Comment:
Not too sure what's happening here, but `log(0)` is undefined. Is that ok?

Reply:
If it ain't broke...